charts/opentelemetry-collector/templates/_pod.tpl

```diff
@@ -148,10 +148,10 @@ initContainers:
 priorityClassName: {{ .Values.priorityClassName | quote }}
 {{- end }}
 volumes:
-  {{- if .Values.configMap.create }}
+  {{- if or .Values.configMap.create .Values.configMap.existingName }}
   - name: {{ include "opentelemetry-collector.lowercase_chartname" . }}-configmap
     configMap:
-      name: {{ include "opentelemetry-collector.fullname" . }}{{ .configmapSuffix }}
+      name: {{ .Values.configMap.existingName | default (printf "%s%s" (include "opentelemetry-collector.fullname" .) (.configmapSuffix)) }}
```

Comment:
Lets move this logic into a template so that we can do a simple if/else

Reply:
Updated c0fd7da296b427a4e2496495a8eb94fae19d82b3 @TylerHelmuth